From b474793c5509c3a68a60e68b5db0745eb821ce1d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 25 Mar 2025 01:51:16 +0100
Subject: [PATCH] Set color scheme later

GTK 4.18 no longer allow to create displays before GDK has been initialized.
Therefore the style manager cannot be initialized before that, as it causing an abort.

See details: https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/7836
---
 src/application.js | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/application.js b/src/application.js
index e051fcc..223d5c0 100644
--- a/src/application.js
+++ b/src/application.js
@@ -27,7 +27,9 @@ export default function Application() {
     );
   }
 
-  Adw.StyleManager.get_default().set_color_scheme(Adw.ColorScheme.FORCE_DARK);
+  application.connect("startup", () => {
+    Adw.StyleManager.get_default().set_color_scheme(Adw.ColorScheme.FORCE_DARK);
+  });
 
   // FIXME: Cannot deal with mailto:, xmpp:, ... URIs
   // GFile mess the URI if the scheme separator does not include "//" like about:blank or mailto:foo@bar.com
